src/Microsoft.AspNet.Mvc.Core/Filters/AuthorizeAttribute.cs

```diff
@@ -14,67 +12,85 @@ namespace Microsoft.AspNet.Mvc
 {
     public class AuthorizeAttribute : AuthorizationFilterAttribute
     {
-        protected Claim[] _claims;
+        private string _roles;
+        private string[] _rolesSplit;
 
-        public AuthorizeAttribute()
-        {
-            _claims = new Claim[0];
-        }
+        public AuthorizeAttribute() { }
 
-        public AuthorizeAttribute([NotNull]IEnumerable<Claim> claims)
+        public AuthorizeAttribute(string policy)
         {
-            _claims = claims.ToArray();
+            Policy = policy;
         }
 
-        public AuthorizeAttribute(string claimType, string claimValue)
-        {
-            _claims = new[] { new Claim(claimType, claimValue) };
-        }
+        public string Policy { get; set; }
 
-        public AuthorizeAttribute(string claimType, string claimValue, params string[] otherClaimValues)
-            : this(claimType, claimValue)
+        public string Roles
         {
-            if (otherClaimValues.Length > 0)
+            get { return _roles; }
+            set
             {
-                _claims = _claims.Concat(otherClaimValues.Select(claim => new Claim(claimType, claim))).ToArray();
+                _roles = value;
+                if (string.IsNullOrWhiteSpace(_roles))
+                {
+                    _rolesSplit = null;
+                }
+                else
+                {
+                    _rolesSplit = _roles.Split(',');
+                }
             }
         }
 
         public override async Task OnAuthorizationAsync([NotNull] AuthorizationContext context)
         {
             var httpContext = context.HttpContext;
-            var user = httpContext.User;
 
-            // when no claims are specified, we just need to ensure the user is authenticated
-            if (_claims.Length == 0)
+            // Allow Anonymous skips all authorization
+            if (HasAllowAnonymous(context))
             {
-                var userIsAnonymous =
-                    user == null ||
-                    user.Identity == null ||
-                    !user.Identity.IsAuthenticated;
+                return;
+            }
 
-                if (userIsAnonymous && !HasAllowAnonymous(context))
+            var authService = GetAuthService(httpContext);
+
+            // Build a policy for the requested roles if specified
+            if (_rolesSplit != null)
+            {
+                var rolesPolicy = new AuthorizationPolicyBuilder();
+                rolesPolicy.RequiresRole(_rolesSplit);
+                if (!await authService.AuthorizeAsync(rolesPolicy.Build(), httpContext, context))
                 {
                     Fail(context);
+                    return;
                 }
```

Comment:
Right now we consider setting Role = "foo" to be an additional requirement to the policy, so if you specify both, you need to be in the Role AND meet the policy.  We can revisit this though, as it is arbitrary